client/shaders/nodes_shader/opengl_vertex.glsl

```diff
@@ -199,24 +201,37 @@ void main(void)
 	vNormal = inVertexNormal;
 
 	// Calculate color.
-	vec4 color = inVertexColor;
+
+	// Extract the hardware color and send that
+	// to the fragment shader for mixing with texture color.
+	hardwareColor.rgb = inVertexColor.rgb;
+
+	// Unpack the values from alpha
```

Comment:
Why is this quantization of just 16 steps for the ratio and light intensity acceptable?

Reply:
That one is my code, so I'll explain.

Since light value has that quantization, here it theoretically could have more values due to interpolation, but in the end effect is very similar, because it's interpolated between vertices anyway.